Source/Core/VideoCommon/PixelEngine.cpp

```diff
@@ -273,26 +251,22 @@ static void SetFinish_OnMainThread(u64 userdata, s64 cyclesLate)
 // THIS IS EXECUTED FROM VIDEO THREAD
 void SetToken(const u16 _token, const int _bSetTokenAcknowledge)
 {
+  INFO_LOG(PIXELENGINE, "VIDEO Backend raises INT_CAUSE_PE_TOKEN (btw, token: %04x)", _token);
+  s_Token.store(_token);
```

Comment:
I've added a thrid patch which passes all tokens by event. But for performance, this is not done in a queue. The events are merged on the GPU thread already. So coretiming will only see a single event.